tests/monitoring/monitoring.go

```diff
@@ -630,8 +636,121 @@ var _ = Describe("[Serial][sig-monitoring]Prometheus Alerts", Serial, decorators
+
+			Eventually(func() string {
+				vh, err := virtClient.AppsV1().DaemonSets(originalKv.Namespace).Get(context.Background(), virtHandler.deploymentName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return vh.Spec.Template.Spec.NodeSelector["kubernetes.io/hostname"]
+			}, 90*time.Second, 5*time.Second).Should(Equal("does-not-exist"))
+
+			Eventually(func() int {
+				vh, err := virtClient.AppsV1().DaemonSets(originalKv.Namespace).Get(context.Background(), virtHandler.deploymentName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return int(vh.Status.NumberAvailable + vh.Status.NumberUnavailable)
+			}, 90*time.Second, 5*time.Second).Should(Equal(0))
+
+			return kv
+		}
+
+		setEmulation := func(newStatus bool) (previousStatus bool) {
+			originalKv := util.GetCurrentKv(virtClient)
+			kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(originalKv.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			previousStatus = kv.Spec.Configuration.DeveloperConfiguration.UseEmulation
+			kv.Spec.Configuration.DeveloperConfiguration.UseEmulation = newStatus
+
+			Eventually(func() error {
+				kv, err = virtClient.KubeVirt(originalKv.Namespace).Update(kv)
+				return err
+			}, 30*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+
+			Eventually(func() bool {
+				kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(originalKv.Name, &metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return kv.Spec.Configuration.DeveloperConfiguration.UseEmulation
+			}, 90*time.Second, 5*time.Second).Should(Equal(newStatus))
+
+			return previousStatus
+		}
+
+		restoreVirtHandler := func(kv *v1.KubeVirt) {
+			originalKv := util.GetCurrentKv(virtClient)
+			kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(originalKv.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			kv.Spec.CustomizeComponents = v1.CustomizeComponents{}
+
+			Eventually(func() error {
+				kv, err = virtClient.KubeVirt(originalKv.Namespace).Update(kv)
+				return err
+			}, 30*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+		}
+
+		It("KubeVirtNoAvailableNodesToRunVMs should be triggered when there are no available nodes in the cluster to run VMs", func() {
+			By("Scaling down virt-handler")
+			kv := disableVirtHandler()
+
+			By("Removing the kubevirt.io/schedulable label from all nodes")
+			nodes, err := virtClient.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			removeStatusAllocatableKVMFromNodes(virtClient, nodes)
+
+			By("Verifying KubeVirtNoAvailableNodesToRunVMs alert exists if emulation is disabled")
+			originalEmulationConfig := setEmulation(false)
+			verifyAlertExistWithCustomTime("KubeVirtNoAvailableNodesToRunVMs", 10*time.Minute)
+
+			By("Restoring virt-handler and emulation")
+			restoreVirtHandler(kv)
+			setEmulation(originalEmulationConfig)
```

Comment:
Is this redundant?

Reply:
I'm making sure the emulation is disabled before with `originalEmulationConfig := setEmulation(false)`
Here I'm ensuring to revert kubevirt resource to the original state

Reply:
Please remove it. The testsuite already does this.